### ISSUE TITLE
Fix table header cell sorting states

### ISSUE THREAD

**Issue by SeamusLeonardHPE** [grommet/hpe-design-system#3601]
Discovery from Marina's table work.

Table header is missing 2 variations, each with 2 states

https://www.figma.com/file/cWWkj83XeKhVu5cX4L8EQE?node-id=2:11143&mode=design#614057500

header-default/sorted + descend
header-default/sorted + ascend
header-numerical/sorted + descend
header-numerical/sorted + ascend

**Comment by SeamusLeonardHPE** [comment]
Created 4 new components in a branch.
Ready for review & merge to master.

@halocline @ashifalinadaf 

https://www.figma.com/file/0wqEYwXoSmC3n5ugZQPhBb/%233601--Add-table-header-states?type=design&node-id=2390-13951&mode=design&t=DO2Xnkabs2XbsfBX-4



<img width="247" alt="image" src="https://github.com/grommet/hpe-design-system/assets/135040188/566c3905-3900-4c28-8d86-6a23a6732c5d">
<img width="254" alt="image" src="https://github.com/grommet/hpe-design-system/assets/135040188/fda38f92-049a-4dd8-a405-9a8ba8cca507">

**Comment by SeamusLeonardHPE** [comment]
Maybe chance to clean out ould deprecated examples of table row/columns?

**Comment by KennyAtHPE** [comment]
@SeamusLeonardHPE so it looks like the four new components are being created using an old deleted component. We need to either restore the deleted component or build this from scratch entirely. I would suggest the later.

Let me know if you would like me to take a stab at rebuilding those components, furthermore, I can help swap out the deprecated examples

**Comment by SeamusLeonardHPE** [comment]
@KennyAtHPE tired to add a comment on figma, but connection seems unstable this morning.

All of the header cells are built using a nested deleted component. I just duplicated & removed the focus state. There are similar issues in multiple other components that I have documented and will be adding tickets.

All table elements need rebuilt in the long term, but I would suggest speed to resolve this issue is more important now as Marinas work work requires these states. Happy to discuss this with the team, but I think we should create a separate ticket to restore/rebuild as follow up work since it would likely involve breaking changes and not help the product team in the short term.
